core/src/main/scala/org/apache/spark/sql/rapids/tool/util/ToolsPlanGraph.scala

```diff
@@ -195,53 +33,30 @@ case class DBReflectionContainer() {
  * Build a SparkPlanGraph from the root of a SparkPlan tree.
  */
 object ToolsPlanGraph {
-  // TODO: We should have a util to detect if the runtime is Databricks.
-  //      This can be achieved by checking for spark properties
-  //      spark.databricks.clusterUsageTags.clusterAllTags
-  private lazy val dbRuntimeReflection = DBReflectionContainer()
   // By default call the Spark constructor. If this fails, we fall back to the DB constructor
   def constructGraphNode(id: Long, name: String, desc: String,
       metrics: collection.Seq[SQLPlanMetric]): SparkPlanGraphNode = {
-    try {
-      new SparkPlanGraphNode(id, name, desc, metrics)
-    } catch {
-      case _: java.lang.NoSuchMethodError =>
-        dbRuntimeReflection.constructNode(id, name, desc, metrics)
-    }
+    GraphReflectionAPIHelper.api.get.constructNode(id, name, desc, metrics)
   }
 
   private def constructSQLPlanMetric(name: String,
       accumulatorId: Long,
       metricType: String): SQLPlanMetric = {
     val accNameRef = AccumNameRef.getOrCreateAccumNameRef(name)
-    try {
-      SQLPlanMetric(accNameRef.value, accumulatorId, metricType)
-    } catch {
-      case _: java.lang.NoSuchMethodError =>
-        dbRuntimeReflection.constructSQLPlanMetric(
-          accNameRef.value, accumulatorId, metricType)
-    }
+    GraphReflectionAPIHelper.api.get.constructSQLPlanMetric(accNameRef.value, accumulatorId,
+      metricType)
   }
 
   private def constructCluster(id: Long,
       name: String,
       desc: String,
       nodes: mutable.ArrayBuffer[SparkPlanGraphNode],
       metrics: collection.Seq[SQLPlanMetric]): SparkPlanGraphCluster = {
-    try {
-      new SparkPlanGraphCluster(id, name, desc, nodes, metrics)
-    } catch {
-      case _: java.lang.NoSuchMethodError =>
-        dbRuntimeReflection.constructCluster(id, name, desc, nodes, metrics)
-    }
+    GraphReflectionAPIHelper.api.get.constructCluster(id, name, desc, nodes, metrics)
```

Comment:
these (GraphReflectionAPIHelper.api) are all Options and we just call get here on it without doing else or checking so is there any reason for that to be an option?

Reply:
I refactored the code to define the "method graphBuilder" as a field depending on the initialization of the APIs